code/modules/atmospherics/machinery/datum_pipeline.dm

```diff
@@ -231,25 +232,28 @@
 			continue
 		gas_mixture_list += pipeline.other_airs
 		gas_mixture_list += pipeline.air
-		for(var/atmosmch in pipeline.other_atmosmch)
-			if (istype(atmosmch, /obj/machinery/atmospherics/components/binary/valve))
-				var/obj/machinery/atmospherics/components/binary/valve/considered_valve = atmosmch
-				if(considered_valve.on)
-					pipeline_list |= considered_valve.parents[1]
-					pipeline_list |= considered_valve.parents[2]
-			else if (istype(atmosmch, /obj/machinery/atmospherics/components/unary/portables_connector))
-				var/obj/machinery/atmospherics/components/unary/portables_connector/considered_connector = atmosmch
-				if(considered_connector.connected_device)
-					gas_mixture_list += considered_connector.connected_device.return_air()
+		for(var/obj/machinery/atmospherics/components/atmosmch as anything in pipeline.other_atmosmch)
+			if(QDELETED(atmosmch))
+				continue
+
+			var/new_pipelines = atmosmch.returnPipenetsForReconcilation(src)
+			if(new_pipelines)
+				pipeline_list |= new_pipelines
+
+			var/new_airs = atmosmch.returnAirsForReconcilation(src)
+			if(new_airs)
+				gas_mixture_list |= new_airs
 
 	var/total_thermal_energy = 0
 	var/total_heat_capacity = 0
 	var/datum/gas_mixture/total_gas_mixture = new(0)
 
 	var/list/total_gases = total_gas_mixture.gases
 
-	for(var/mixture in gas_mixture_list)
-		var/datum/gas_mixture/gas_mixture = mixture
+	for(var/datum/gas_mixture/gas_mixture as anything in gas_mixture_list)
+		if(!gas_mixture)
+			continue
+
```

Comment:
Why the check? if a null ends up in this list that means someone is either fucking with it, or qdel'd a gasmixture, and either way I want to know

Reply:
Leftover from experimenting with solutions, it shouldn't be here